flake.nix

```diff
@@ -0,0 +1,43 @@
+{
+  description = "Lychee - A fast, async, stream-based link checker";
+
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
+    flake-utils.url = "github:numtide/flake-utils";
+    fenix.url = "github:nix-community/fenix";
+  };
+
+  outputs = {
+    self,
+    nixpkgs,
+    flake-utils,
+    fenix,
+    ...
+  }:
+    flake-utils.lib.eachDefaultSystem (system: let
+      pkgs = import nixpkgs {inherit system;};
+      toolchain = fenix.packages.aarch64-darwin.fromToolchainFile {
+        file = ./rust-toolchain.toml;
+        sha256 = "sha256-yMuSb5eQPO/bHv+Bcf/US8LVMbf/G/0MSfiPwBhiPpk=";
```

Comment:
What exactly is this SHA pinning?

Reply:
That’s a common pattern in nix documentation. Until you figure out the hash of the file you’re referencing, you can put that fake hash (personally I just put an empty hash, works the same) and then you will get a mismatch when you actually try to evaluate the expression. Then you can just copy paste the hash that it actually gets and you’re good to go. There is also a mechanism to compute it ahead of time, don’t know it off the top of my head

Reply:
We can also go with rust-overlay as @thomas-zahner did in his PR, or even easier not pin the Rust version at all and just take what nixpkgs gives us (as long as you're not living on the bleeding edge of Rust releases, that will work out fine). It's really up to the maintainers here if they're comfortable having to update the hash whenever the toolchain config changes.